sympy/solvers/solveset.py

```diff
@@ -847,6 +847,9 @@ def solveset(f, symbol=None, domain=S.Complexes):
 
     if not free_symbols:
         b = Eq(f, 0)
+        if isinstance(symbol, Number):
```

Comment:
Doing this won't pass the following test case, it gives the message as
```
File "c:\users\dell\desktop\sympy\sympy\solvers\tests\test_solveset.py", line 880, in test_solveset
assert solveset(0, domain=S.Reals) == S.Reals
File "C:\Users\DELL\Desktop\sympy\sympy\solvers\solveset.py", line 854, in solveset
multivariate equation.'''))
ValueError:
The independent variable must be specified for a multivariate
equation.

 tests finished: 345 passed, 21 skipped, 32 expected to fail, 1 exceptions,
in 327.12 seconds
DO *NOT* COMMIT!

```
The reason for placing 
```
if isinstance(symbol, Number):
        raise ValueError('A Symbol must be given, not type %s: %s' %
                 (type(symbol), symbol))
```
after Line 848 was that `solveset` only had to check for cases where the `expression` does not contains any `free symbols` and the `symbol` is a `number` , because the cases where the `expression` contains `free symbols` and `symbol` is a `number` are already handled by `ValueError` in Line 866. 
This is what I thought, I might be wrong though..